src/formatDate.js

```diff
@@ -9,6 +9,64 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const dateSeparated = date.split(fromFormat[3]);
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i < 3; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        year = dateSeparated[i];
+        break;
+      case 'YY':
+        year = dateSeparated[i];
+        break;
+      case 'MM':
+        month = dateSeparated[i];
+        break;
+      case 'DD':
+        day = dateSeparated[i];
+        break;
+    }
+  }
+
+  let finalDate = '';
+
+  for (let i = 0; i < 3; i++) {
+    switch (toFormat[i]) {
+      case 'MM':
+        finalDate += month;
+        break;
+      case 'DD':
+        finalDate += day;
+        break;
+      case 'YY':
+        if (year.length === 2) {
+          finalDate += year;
+        } else {
+          finalDate += year.slice(-2);
```

Comment:
The check for 'year.length === 2' inside the 'YY' case is unnecessary since the 'fromFormat' has already determined the length of the 'year' variable. You can directly use 'year.slice(-2)' without the check.